core/lib/server/form.ts

```diff
@@ -215,61 +207,23 @@ export const createFormInviteLink = async (props: FormInviteLinkProps) => {
 	return magicLink;
 };
 
-const componentsBySchemaTable = Object.entries(componentsBySchema)
-	.map(([schema, components]) => {
-		const component = components[0]
-			? `'${components[0]}'::"InputComponent"`
-			: `null::"InputComponent"`;
-		return `('${schema}'::"CoreSchemaType", ${component})`;
-	})
-	.join(", ");
-
 export const insertForm = (
-	pubTypeId: PubTypesId,
+	pubType: GetPubTypesResult[number],
 	name: string,
 	slug: string,
 	communityId: CommunitiesId,
 	isDefault: boolean,
 	trx = db
 ) => {
+	const ranks = mudder.base62.mudder(pubType.fields.length + 1);
```

Comment:
Because we need to generate these ranks in js, I changed this function to expect a full pubType as an input, which means an additional query + roundtrip to the db before the insert. I think that's a fine tradeoff with how much simpler this looks now! But if we someday find this is too costly, mudder has no dependencies, so we could probably run it [inside postgres](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/PostgreSQL.Concepts.General.UpgradingPLv8.html).

Reply:
it's definitely not a problem for form inserts. but with pub inserts/updates i could see the roundtrip for finding the highest existing rank becoming a performance issue. mostly it's just that it sounds sick though 😄